tests/integration/test_cross_model_remote_write.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+
+import pytest
+from helpers import check_prometheus_is_ready, oci_image, run_promql
+from juju.controller import Controller  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+prometheus_name = "prometheus"
+agent_name = "grafana-agent"
+
+
+@pytest.mark.abort_on_fail
+async def test_create_remote_write_models(ops_test, prometheus_charm):
+    """Test that Prometheus can be related with the Grafana Agent over remote_write."""
+    await ops_test.model.deploy(
+        prometheus_charm,
+        resources={"prometheus-image": oci_image("./metadata.yaml", "prometheus-image")},
+        application_name=prometheus_name,
+    )
+
+    # pytest_operator keeps a dict[str, ModelState] for internal reference, and they'll
+    # all get cleaned up just like the automatic one. The alias for the first one is
+    # 'main' if we want to get it.
+    #
+    # 'consumer' is an alias, not the name of the model
+    await ops_test.track_model("consumer")
+
+    offer, consumer = ops_test.models.get("main"), ops_test.models.get("consumer")
+    await consumer.model.deploy(
+        "grafana-agent-k8s",
+        application_name=agent_name,
+        channel="edge",
+    )
+
+    await asyncio.gather(
+        offer.model.wait_for_idle(apps=[prometheus_name], status="active"),
+        consumer.model.wait_for_idle(apps=[agent_name], status="active"),
+    )
+
+    assert check_prometheus_is_ready(ops_test, prometheus_name, 0)
```

Comment:
We should probably at some point refactor many of the helpers to take a model (or a ModelState?) instead of `ops_test`.

Reply:
Eventually, maybe... if we don't rewrite it. For now, it's only used here.